testutil/checkers.go

```diff
@@ -83,3 +83,61 @@ func (c *containsChecker) Check(params []interface{}, names []string) (result bo
 		return false, fmt.Sprintf("%T is not a supported container", container)
 	}
 }
+
+type deepContainsChecker struct {
+	*check.CheckerInfo
+}
+
+// DeepContains is a Checker that looks for a elem in a container using
+// DeepEqual.  The elem can be any object. The container can be an array, slice
+// or string.
+var DeepContains check.Checker = &deepContainsChecker{
+	&check.CheckerInfo{Name: "DeepContains", Params: []string{"container", "elem"}},
+}
+
+func (c *deepContainsChecker) Check(params []interface{}, names []string) (result bool, error string) {
+	defer func() {
+		if v := recover(); v != nil {
```

Comment:
In fact, there's a lot of logic here and above which is exactly the same. Perhaps these two issues may be solved in one go, by factoring comparison out of the overall problem and into a parameter that may be passed in.